include/sys/time_units.h

```diff
@@ -11,6 +11,19 @@
 extern "C" {
 #endif
 
+/** @brief System-wide macro to denote "forever" in milliseconds
+ *
+ *  Usage of this macro is limited to APIs that want to expose a timeout value
+ *  that can optionally be unlimited, or "forever".
+ *  This macro can not be fed into kernel functions or macros directly. Use
+ *  @ref SYS_MSEC instead.
+ */
+#define SYS_FOREVER_MS (-1)
+
+/** @brief System-wide macro to convert milliseconds to kernel timeouts
+ */
+#define SYS_MSEC(ms) (ms == SYS_FOREVER_MS ? K_FOREVER : K_MSEC(ms))
```

Comment:
I would rather see this as `SYS_MS_TIMEOUT` since it's producing a timeout from a millisecond value.  There's nothing in the name that suggests the result is not milliseconds, and there's no pre-existing practice like with `K_MSEC()` to preserve by trying to match that naming convention.

Reply:
I think we need to put () over ms so `#define SYS_MSEC(ms) ((ms) == SYS_FOREVER_MS ? K_FOREVER : K_MSEC(ms))`

Reply:
Btw, why MS and MSEC are being used, couldn't we stick to one? (MS is fine imo)